JotunnLib/Utils/ModCompatibility/ModModule.cs

```diff
@@ -19,6 +21,7 @@ public ModModule(string name, System.Version version, CompatibilityLevel compati
 
         public ModModule(ZPackage pkg)
         {
+            guid = pkg.ReadString();
```

Comment:
this will cause an error when an old Jotunn version connects to this version or vice versa (because ZPackage is very primitive and just reads data from the stream). Unfortunately, ModModule has no int version field for the data layout itself.

I'm rather reluctant to introduce such an issue (especially since it breaks the disconnect UI), after a Jotunn update the connections with mismatching versions is a lot higher. I think the only non-breaking way would be to add a new RPC first, with the changed data layout. If the new RPC is received, the second (old one) can be ignored. And connection is refused either way because the version missmatches

Reply:
Oh, thanks for letting me know about this. I haven't dealt with RPCs much before and haven't run into issues with different versions of the same mod connecting before so I didn't think of that at all. I'm not entirely sure what you mean by adding a new RPC but I'll take a look when I have time over the next week and see if I can figure it out.